### ISSUE TITLE
Cannot update node_modules when using cache with workspace:* because pnpm-lock is not updated

### ISSUE THREAD

**Issue by imrim12** [pnpm/pnpm#6655]
### Code to reproduce the issue:
When using `workspace:*` in a monorepo, the packages got linked together nicely.

But since we apply caching for the `node_modules` to reduce the installation time, it has update issue, the `pnpm-lock` is not updated, therefore the cache hash is also not changed.

### Expected behavior:
The `pnpm-lock` got some extra info about the package like its version in order for it to be changed when the package got bumped => the hash is changed and the cache is cleared

```
      '@mycompany/utils':
        version: 1.0.0-dev.123 <= **add this**
        dependencies:
        devDependencies:
```

### Actual behavior:
The `pnpm-lock` stays the same
```
      '@mycompany/utils':
        dependencies:
        devDependencies:
```

### Additional information:

I have thought about using a proper version but it would be too much work, when you have the bump the package version, and then every where it got used, for a utils package for example, this would require a lot of scanning and changing, i want to have more organizing on my project so there're a bunch of nested folder, which make it hard for some auto-bump Github Actions on the marketplace

I hope to look for a better solution, thanks

**Comment by zkochan** [comment]
I don't think that this is a good idea because it will increase the amount of git conflicts.

Also, I don't understand why a bump to the project would invalidate the node_modules cache. Workspace projects are not inside node_modules, they are just symlinked to node_modules.
